dns/proxy_test.go

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright (C) 2020 The "MysteriumNetwork/node" Authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package dns
+
+import (
+	"net"
+	"testing"
+
+	"github.com/miekg/dns"
+	"github.com/mysteriumnetwork/node/core/policy"
+	"github.com/mysteriumnetwork/node/firewall"
+	"github.com/mysteriumnetwork/node/market"
+	"github.com/stretchr/testify/assert"
+)
+
+var (
+	policyDNSZone      = market.AccessPolicy{ID: "wildcard-domain"}
+	policyDNSZoneRules = market.AccessPolicyRuleSet{
+		ID: "wildcard-domain",
+		Allow: []market.AccessRule{
+			{Type: market.AccessPolicyTypeDNSZone, Value: "wildcard.com"},
+		},
+	}
+
+	policyDNSHostname      = market.AccessPolicy{ID: "domain"}
+	policyDNSHostnameRules = market.AccessPolicyRuleSet{
+		ID: "domain",
+		Allow: []market.AccessRule{
+			{Type: market.AccessPolicyTypeDNSHostname, Value: "single.com"},
+		},
+	}
+)
+
+func Test_Proxy_WhitelistsAllowedHostnames(t *testing.T) {
+	mockedBlocker := &trafficBlockerMock{
+		allowIPCalls: map[string]int{},
+	}
+	proxy := NewProxy("localhost", 10053, mockedBlocker, createPolicies())
+
+	err := proxy.whitelistByAnswer(&dns.Msg{
```

Comment:
It can be simplified using a test table.

Reply:
added